packages/components/src/InputPhoneNumber/InputMask.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactElement, cloneElement, useState } from "react";
+import styles from "./InputMask.css";
+import { FormFieldProps } from "../FormField";
+
+export interface InputMaskProps {
+  /**
+   * Determines the masking pattern. By default, a `*` is used to be the value
+   * you're expecting the user would type.
+   */
+  readonly pattern: string;
+
+  /**
+   * Change the delimiter when you need to have a `*` as a value that the input
+   * returns.
+   *
+   * @default "*"
+   */
+  readonly delimiter?: string;
+
+  /**
+   * Allow/prevent users adding from more value than the desired pattern.
+   * For example, your pattern could accept 10 characters. If it's strict, your
+   * users can only type 10 characters.
+   *
+   * @default true
+   */
+  readonly strict?: boolean;
+
+  readonly children: ReactElement<FormFieldProps>;
+}
+
+export function InputMask({
+  children,
+  delimiter = "*",
+  pattern,
+  strict = true,
+}: InputMaskProps) {
+  const { value: inputValue, onChange } = children.props;
+  const [hasMask, setHasMask] = useState(!inputValue);
```

Comment:
I think I understand what this line is doing (well enough, lol) but a comment here might make this a bit quicker for devs to understand.